pkg/cluster/kubeadm/config.go

```diff
@@ -101,6 +105,48 @@ kind: InitConfiguration
 # we use a random local port for the API server
 apiEndpoint:
   bindPort: {{.APIBindPort}}
+---
+# no-op entry that exists soley so it can be patched
+apiVersion: kubeadm.k8s.io/v1alpha3
+kind: JoinConfiguration
+---
+# no-op entry that exists soley so it can be patched
+apiVersion: kubelet.config.k8s.io/v1alpha1
+kind: KubeletConfiguration
+---
+# no-op entry that exists soley so it can be patched
+apiVersion: kubeproxy.config.k8s.io/v1alpha1
+kind: KubeProxyConfiguration
+`
+
+// ConfigTemplateBetaV1 is the kubadm config template for API version v1beta1
+const ConfigTemplateBetaV1 = `# config generated by kind
+apiVersion: kubeadm.k8s.io/v1beta1
+kind: ClusterConfiguration
+kubernetesVersion: {{.KubernetesVersion}}
+clusterName: {{.ClusterName}}
+# on docker for mac we have to expose the api server via port forward,
+# so we need to ensure the cert is valid for localhost so we can talk
+# to the cluster after rewriting the kubeconfig to point to localhost
+apiServerCertSANs: [localhost]
```

Comment:
it's has changed in v1beta1:
```
apiSever:
  certSANs: ....
```

kustomize is confusing me.

Reply:
thanks

which part is confusing? we aren't using it here

Reply:
fixed this